staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -330,17 +338,20 @@ func (c *AvailableConditionController) sync(key string) error {
 	availableCondition.Reason = "Passed"
 	availableCondition.Message = "all checks passed"
 	apiregistration.SetAPIServiceCondition(apiService, availableCondition)
-	_, err = updateAPIServiceStatus(c.apiServiceClient, originalAPIService, apiService)
+	_, err = updateAPIServiceStatus(c.apiServiceClient, c.localServiceStatusUpdater, originalAPIService, apiService)
 	return err
 }
 
 // updateAPIServiceStatus only issues an update if a change is detected.  We have a tight resync loop to quickly detect dead
 // apiservices.  Doing that means we don't want to quickly issue no-op updates.
-func updateAPIServiceStatus(client apiregistrationclient.APIServicesGetter, originalAPIService, newAPIService *apiregistration.APIService) (*apiregistration.APIService, error) {
+func updateAPIServiceStatus(client apiregistrationclient.APIServicesGetter, localServiceStatusUpdater LocalAPIServiceStatusUpdater, originalAPIService, newAPIService *apiregistration.APIService) (*apiregistration.APIService, error) {
 	if equality.Semantic.DeepEqual(originalAPIService.Status, newAPIService.Status) {
 		return newAPIService, nil
 	}
 
+	if localServiceStatusUpdater != nil {
+		localServiceStatusUpdater.UpdateAPIServiceStatus(newAPIService)
```

Comment:
I wouldn't guard this call with the DeepEqual above, but always go to the local availability updater. Then check inside of `localServiceStatusUpdater.UpdateAPIServiceStatus` whether the value has changed. I fear otherwise that we might see races when `localServiceStatusUpdater.UpdateAPIServiceStatus` is a noop because the handler info atmoic var is still unset.